packages/babel-parser/src/plugins/typescript.js

```diff
@@ -501,13 +501,24 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       const node: N.TsTupleType = this.startNode();
       node.elementTypes = this.tsParseBracketedList(
         "TupleElementTypes",
-        this.tsParseType.bind(this),
+        this.tsParseTupleElementType.bind(this),
         /* bracket */ true,
         /* skipFirstToken */ false,
       );
       return this.finishNode(node, "TSTupleType");
     }
 
+    tsParseTupleElementType(): N.TsType {
+      // Maybe necessary for optional type
+      const wrapperNode = this.startNode();
+      const type = this.tsParseType();
+      if (this.eat(tt.question)) {
+        wrapperNode.typeAnnotation = type;
```

Comment:
We should create `wrapperNode` inside this condition. You can use `this.startNodeAtNode(type)`

Reply:
👍 Fixed.